repositories/EventRepository.ts

```diff
@@ -41,8 +41,14 @@ export class EventRepository extends BaseRepository<EventModel> {
   }
 
   public async isUnusedAttendanceCode(attendanceCode: string): Promise<boolean> {
-    const count = await this.repository.count({ attendanceCode });
-    return count === 0;
+    const attendanceCodeDuplicates = await this.repository.find({
+      where: {
+        attendanceCode,
+        end: MoreThanOrEqual(new Date()),
```

Comment:
Technically, checking `new Date()` is a more strict check. What's your opinion on comparing the new event's start date with existing events' end dates? Not saying this approach is wrong but I'm just curious if you have any edge cases in mind

Reply:
An edge case occurs if two events with the same attendance code were both being planned for the future. I will edit this method to account for this case